### PR TITLE
Update DomainInfo model to reflect RFC

### LINKED ISSUE #31: DomainInfo should accept authInfo data
https://datatracker.ietf.org/doc/html/rfc5731#section-3.1.2

e.g.,

```
<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<epp xmlns="urn:ietf:params:xml:ns:epp-1.0">
    <command>
        <info>
            <domain:info xmlns:domain="urn:ietf:params:xml:ns:domain-1.0">
                <domain:name hosts="all">example.com</domain:name>
                <domain:authInfo>
                    <domain:pw>2fooBAR</domain:pw>
                </domain:authInfo>
            </domain:info>
        </info>
        <clTRID>ABC-12345</clTRID>
    </command>
</epp>
```

### PR DESCRIPTION
**Pull request by nrempel**
Closes #31

This pull request updates several of the fields in DomainInfo to be optional. It also updates DomainInfo to accept an optional auth_password parameter.

This pull request includes #30 and all of its children.

### CODE REVIEW

File: epp-client/src/tests/de.rs [hunk 1]
```diff
@@ -302,59 +302,59 @@ mod response {
         let ns_list = result.info_data.ns.as_ref().unwrap();
         let ns = (*ns_list).host_obj.as_ref().unwrap();
         let hosts = result.info_data.hosts.as_ref().unwrap();
+        let statuses = result.info_data.statuses.as_ref().unwrap();
```
Comment (djc) [inline epp-client/src/tests/de.rs:305]: If the previous commit doesn't pass tests without this commit, these should probably be squashed?

Comment (nrempel) [reply epp-client/src/tests/de.rs:302]: Fair point. Squashed.

**Comment by masalachai** [comment]
@nrempel Could you rebase this again and let me know? I'm online right now so we can merge and rebase these PRs one by one.

**Comment by nrempel** [comment]
> @nrempel Could you rebase this again and let me know? I'm online right now so we can merge and rebase these PRs one by one.
> 
> EDIT: Actually, seems like you still have some discussion ongoing on #39. And the other PRs depend on that changes in that one. So just rebase this one and we can merge it.

Thanks, rebased

**Comment by masalachai** [comment]
> Edit: fixing build issues... just a moment

I've fixed the failing doctest in consolidate.rs in the `main` branch so should just need a rebase again here.

**Comment by nrempel** [comment]
Great thank you. I think should be good once CI passes.

**Comment by masalachai** [comment]
Thanks. Merging this now.

**Pull request merged**